### PR TITLE
PyConsoleCopyHandler fixes

### PR DESCRIPTION
**Pull request by da1z**
- fixed extra \n appended when copying from console
- fixed columnar selection in console

### CODE REVIEW

File: python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt [hunk 1]
```diff
@@ -35,13 +35,13 @@ class PyConsoleCopyHandler(val originalHandler: EditorActionHandler) : EditorAct
...
   override fun doExecute(editor: Editor, caret: Caret?, dataContext: DataContext) {
     if (!RichCopySettings.getInstance().isEnabled) {
-      return originalHandler.execute(editor, null, dataContext);
+      return originalHandler.execute(editor, null, dataContext)
     }
-    if (true != editor.getUserData(ConsoleViewUtil.EDITOR_IS_CONSOLE_HISTORY_VIEW)) {
-      return originalHandler.execute(editor, null, dataContext);
-
+    if (true != editor.getUserData(ConsoleViewUtil.EDITOR_IS_CONSOLE_HISTORY_VIEW)
+        || editor.caretModel.allCarets.size != 1) {
```
Comment (traff) [inline python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt]: Could you please explain what does this caret size comparison do?

File: python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt [hunk 2]
```diff
@@ -66,7 +66,9 @@ class PyConsoleCopyHandler(val originalHandler: EditorActionHandler) : EditorAct
       val rangeEnd = Math.min(document.getLineEndOffset(i), end)
       if (rangeStart < rangeEnd) {
         sb.append(document.getText(TextRange(rangeStart, rangeEnd)))
-        sb.append("\n")
+        if (rangeEnd < end) {
+          sb.append("\n")
```
Comment (traff) [inline python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt]: If it is an extra \n why not just to trim it and then add \n? I don't quite understand this check.

File: python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt (hunk 1 above)
Comment (fitermay) [inline python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt]: This fixes columnar selection. This basically skips this handler when in column selection mode. In row selection mode there is a single caret and in column selection mode there are multiple.

File: python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt (hunk 2 above)
Comment (fitermay) [inline python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt]: Yes, trim would accomplish the same thing. The only difference would be if the last selected character is a real '\n' .Not sure if it's actually possible.

**Comment by traff** [comment]
Merged.

**Pull request closed without merging**